01_fib/alanhillgemann/main.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+func fib(n int) {
+	if n < 0 {
+		log.SetFlags(0)
+		log.Printf("Fib of %d is not available. Defaulting to minimum fib of 0", n)
+		n = 0
+	} else if n > 45 {
+		log.SetFlags(0)
+		log.Printf("Fib of %d is not available. Defaulting to maximum fib of 45", n)
+		n = 45
+	}
+	fibSlice := []int{}
+	nextFib := 0
+	for i := 0; i <= n; i++ {
+		switch len(fibSlice) {
```

Comment:
you are effectively switching on `i`.
pull the first two cases out of the loops so you don't have to execute the conditional on every iteration:
```go
fibSlice[0] = 0
fibSlice[1] = 1
for i := 2; i <= n; i++ {
```

Reply:
Fixed